### PR TITLE
doc: update 3 hooks

### PR DESCRIPTION
**Pull request by okhaimie-dev**
## Changes

- Updated the documentation for 3 hooks
  - useContractWrite
  - useSignTypedData
  - useStarkName

### CODE REVIEW

File: website/content/hooks/mutation/useSignTypedData.mdx [hunk 1]
```diff
@@ -7,14 +7,28 @@ hookType: mutation
 
 Hook to disconnect the currently connected wallet.
```
Comment (fracek) [inline website/content/hooks/mutation/useSignTypedData.mdx:8]:
```suggestion
Request user to sign a piece of data.
```

File: website/content/hooks/mutation/useSignTypedData.mdx [hunk 2]
```diff
@@ -7,14 +7,28 @@ hookType: mutation
...
 
+<a href="https://gist.github.com/okhaimie-dev/5bd0ee3eab8b2b76d016f5522569820f" target="_blank">Example Json Data</a>
```
Comment (fracek) [inline website/content/hooks/mutation/useSignTypedData.mdx:10]:
```suggestion
You can read more about Starknet signatures and the meaning of the arguments [on the Starknet.js documentation](https://www.starknetjs.com/docs/guides/signature/).
```

File: website/content/hooks/mutation/useSignTypedData.mdx [hunk 3]
```diff
@@ -7,14 +7,28 @@ hookType: mutation
...
+  const { account } = useAccount();
+  const { data, isPending, signTypedData } = useSignTypedData(exampleData);
+
+	return (
+		<button
+			className="w-full"
+			onClick={() => signTypedData(exampleData)}
+			disabled={!account}
+		>
+			{isPending ? <p>Loading...</p> : <p>Sign Message</p>}
```
Comment (fracek) [inline website/content/hooks/mutation/useSignTypedData.mdx:29]:
```suggestion
			{isPending ? <p>Waiting for wallet...</p> : <p>Sign Message</p>}
```

**Review by fracek — CHANGES_REQUESTED** [review]

**Comment by okhaimie-dev** [comment]
@fracek I committed the suggestions. Let me know if there is anything else to get to the finish line.

**Review by fracek — APPROVED** [review]
Looks good! Thank you!

**Pull request merged**